pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -127,6 +129,9 @@ stdenv.mkDerivation (finalAttrs: {
     ++ lib.optionals capstoneSupport [ capstone ]
     ++ lib.optionals (!userOnly) [ dtc ];
 
+  # a private dependency of PAM which is not linked explicitly in static builds
+  env.NIX_LDFLAGS = lib.optionalString (stdenv.hostPlatform.isStatic && stdenv.isLinux) " -laudit ";
```

Comment:
That sounds like a pkg-config problem we should fix in pam (or in QEMU's meson.build if it's not using pkg-config).